tests/components/dynalite/test_bridge.py

```diff
@@ -1,136 +1,81 @@
 """Test Dynalite bridge."""
-from unittest.mock import Mock, call, patch
+from unittest.mock import Mock, call
 
+from asynctest import patch
 from dynalite_lib import CONF_ALL
 import pytest
 
-from homeassistant.components.dynalite import DATA_CONFIGS, DOMAIN
-from homeassistant.components.dynalite.bridge import BridgeError, DynaliteBridge
+from homeassistant.components import dynalite
 
-from tests.common import mock_coro
 
-
-async def test_bridge_setup():
-    """Test a successful setup."""
-    hass = Mock()
-    entry = Mock()
-    host = "1.2.3.4"
-    entry.data = {"host": host}
-    hass.data = {DOMAIN: {DATA_CONFIGS: {host: {}}}}
-    dyn_bridge = DynaliteBridge(hass, entry)
-
-    with patch.object(
-        dyn_bridge.dynalite_devices, "async_setup", return_value=mock_coro(True)
-    ):
-        assert await dyn_bridge.async_setup() is True
-
-    forward_entries = set(
-        c[1][1] for c in hass.config_entries.async_forward_entry_setup.mock_calls
-    )
-    hass.config_entries.async_forward_entry_setup.assert_called_once()
-    assert forward_entries == set(["light"])
-
-
-async def test_invalid_host():
-    """Test without host in hass.data."""
+@pytest.fixture
+def dyn_bridge():
+    """Define a basic mock bridge."""
     hass = Mock()
-    entry = Mock()
     host = "1.2.3.4"
-    entry.data = {"host": host}
-    hass.data = {DOMAIN: {DATA_CONFIGS: {}}}
+    bridge = dynalite.DynaliteBridge(hass, {dynalite.CONF_HOST: host})
+    return bridge
 
-    dyn_bridge = None
-    with pytest.raises(BridgeError):
-        dyn_bridge = DynaliteBridge(hass, entry)
-    assert dyn_bridge is None
 
+async def test_update_device(dyn_bridge):
+    """Test a successful setup."""
+    async_dispatch = Mock()
 
-async def test_add_devices_then_register():
+    with patch(
+        "homeassistant.components.dynalite.bridge.async_dispatcher_send", async_dispatch
+    ):
+        dyn_bridge.update_device(CONF_ALL)
+        async_dispatch.assert_called_once()
+        assert async_dispatch.mock_calls[0] == call(
+            dyn_bridge.hass, f"dynalite-update-{dyn_bridge.host}"
+        )
+        async_dispatch.reset_mock()
+        device = Mock
+        device.unique_id = "abcdef"
+        dyn_bridge.update_device(device)
+        async_dispatch.assert_called_once()
+        assert async_dispatch.mock_calls[0] == call(
+            dyn_bridge.hass, f"dynalite-update-{dyn_bridge.host}-{device.unique_id}"
+        )
+
+
+async def test_add_devices_then_register(dyn_bridge):
     """Test that add_devices work."""
-    hass = Mock()
-    entry = Mock()
-    host = "1.2.3.4"
-    entry.data = {"host": host}
-    hass.data = {DOMAIN: {DATA_CONFIGS: {host: {}}}}
-    dyn_bridge = DynaliteBridge(hass, entry)
-
+    # First test empty
+    dyn_bridge.add_devices_when_registered([])
+    assert not dyn_bridge.waiting_devices
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5051600)